fast64_internal/sm64/sm64_objects.py

```diff
@@ -1403,6 +1421,719 @@ def draw_act(self, obj, layout, value):
+        if self.bhv_args:
+            return [getattr(self, field) for field, arg_name in zip(self.arg_fields, self.bhv_args)]
+
+    def get_inherit_args(self, context, props):
+        if self.macro not in self._inheritable_macros:
+            return self.macro_args
+        if self.macro == "SET_MODEL":
+            return props.model_id_define
+        if self.macro == "LOAD_COLLISION_DATA":
+            return props.collision_name
+        return self.macro_args
+
+    def draw(self, layout, index):
+        box = layout.box()
+        box.prop(
+            self,
+            "expand",
+            text=f"Bhv Cmd:   {self.macro}",
+            icon="TRIA_DOWN" if self.expand else "TRIA_RIGHT",
+        )
+        if self.expand:
+            prop_split(box, self, "macro", "Behavior Macro")
+            if self.macro in self._inheritable_macros:
+                box.prop(self, "inherit_from_export")
+            if self.macro == "Custom":
+                prop_split(box, self, "num_args", "Num Arguments")
+                for j in range(self.num_args):
+                    prop_split(box, self, f"arg_{j + 1}", f"arg_{j + 1}")
+            elif self.bhv_args and not self.inherit_from_export:
+                for field, arg_name in zip(self.arg_fields, self.bhv_args):
+                    prop_split(box, self, field, arg_name)
+            row = box.row()
+            row.operator("bone.add_behavior_script", text="Add Bhv Cmd").option = index + 1
+            row.operator("bone.remove_behavior_script", text="Remove Bhv Cmd").option = index
+
+
+class SM64_ExportCombinedObject(ObjectDataExporter):
+    bl_idname = "object.sm64_export_combined_object"
+    bl_label = "SM64 Combined Object"
+
+    def write_file_lines(self, path, file_lines):
+        with open(path, "w") as file:
+            [file.write(line) for line in file_lines]
+
+    # exports the model ID load into the appropriate script.c location
+    def export_script_load(self, context, props):
+        # check if model_ids.h exists
+        decomp_path = Path(bpy.path.abspath(bpy.context.scene.decompPath))
+        if props.export_header_type == "Level":
+            script_path = decomp_path / "levels" / f"{props.export_level_name}" / "script.c"
+            self.export_level_specific_load(script_path, props)
+        else:
+            script_path = decomp_path / "levels" / "scripts.c"
+            self.export_group_script_load(script_path, props)
+
+    # delims to notify for when to start and end for sig/alt
+    # if match line, then write out to that line
+    # elif fast64_sig, insert after that line
+    # else insert after alt_condition
+    def find_export_lines(
+        self, file_lines, match_str=None, fast64_signature=None, alt_condition=None, start_delim=None, end_delim=None
+    ):
+        search_sig = False if start_delim else True
+        insert_line = 0
+        alt_insert_line = 0
+        match_line = 0
+        for j, line in enumerate(file_lines):
```

Comment:
Is there any reason you use j over i for loops? Typically you use j for a second index and i for the first index

Reply:
habits from electrical studies no real reason